### ISSUE TITLE
build failure in ISTLSolver.hpp on GCC 8

### ISSUE THREAD

**Issue by andlaus** [OPM/opm-simulators#1574]
when trying to compile flow with GCC 8, I get the following error:

```

cd ~/src/opm-simulators/build-cmake/; make flow
[  0%] Building CXX object CMakeFiles/opmsimulators.dir/opm/autodiff/NewtonIterationBlackoilInterleaved.cpp.o
In file included from /home/and/src/opm-simulators/build-cmake/fake-src/opm/autodiff/NewtonIterationBlackoilInterleaved.cpp:39:
/home/and/src/opm-simulators/build-cmake/fake-src/opm/autodiff/ISTLSolver.hpp: In instantiation of ‘void Opm::ISTLSolver<MatrixBlockType, VectorBlockType, pressureIndex>::constructAMGPrecond(MatrixOperator&, const POrComm&, std::unique_ptr<AMG>&, std::unique_ptr<MatrixOperator>&, double, Opm::MILU_VARIANT) const [with C = Dune::Amg::CoarsenCriterion<Dune::Amg::SymmetricCriterion<Dune::BCRSMatrix<Dune::MatrixBlock<double, 1, 1>, std::allocator<Dune::MatrixBlock<double, 1, 1> > >, Dune::Amg::Diagonal<0> > >; MatrixOperator = Dune::OverlappingSchwarzOperator<Dune::BCRSMatrix<Dune::MatrixBlock<double, 1, 1>, std::allocator<Dune::MatrixBlock<double, 1, 1> > >, Dune::BlockVector<Dune::FieldVector<double, 1> >, Dune::BlockVector<Dune::FieldVector<double, 1> >, Dune::OwnerOverlapCopyCommunication<int> >; POrComm = Dune::OwnerOverlapCopyCommunication<int>; AMG = Opm::BlackoilAmg<Dune::OverlappingSchwarzOperator<Dune::BCRSMatrix<Dune::MatrixBlock<double, 1, 1>, std::allocator<Dune::MatrixBlock<double, 1, 1> > >, Dune::BlockVector<Dune::FieldVector<double, 1> >, Dune::BlockVector<Dune::FieldVector<double, 1> >, Dune::OwnerOverlapCopyCommunication<int> >, Opm::ParallelOverlappingILU0<Dune::BCRSMatrix<Dune::MatrixBlock<double, 1, 1>, std::allocator<Dune::MatrixBlock<double, 1, 1> > >, Dune::BlockVector<Dune::FieldVector<double, 1> >, Dune::BlockVector<Dune::FieldVector<double, 1> >, Dune::OwnerOverlapCopyCommunication<int> >, Dune::Amg::CoarsenCriterion<Dune::Amg::SymmetricCriterion<Dune::BCRSMatrix<Dune::MatrixBlock<double, 1, 1>, std::allocator<Dune::MatrixBlock<double, 1, 1> > >, Dune::Amg::Diagonal<0> > >, Dune::OwnerOverlapCopyCommunication<int>, 0>; MatrixBlockType = Dune::MatrixBlock<double, 1, 1>; VectorBlockType = Dune::FieldVector<double, 1>; int pressureIndex = 0]’:
/home/and/src/opm-simulators/build-cmake/fake-src/opm/autodiff/ISTLSolver.hpp:168:51:   required from ‘void Opm::ISTLSolver<MatrixBlockType, VectorBlockType, pressureIndex>::constructPreconditionerAndSolve(LinearOperator&, Opm::ISTLSolver<MatrixBlockType, VectorBlockType, pressureIndex>::Vector&, Opm::ISTLSolver<MatrixBlockType, VectorBlockType, pressureIndex>::Vector&, const POrComm&, Dune::InverseOperatorResult&) const [with Dune::SolverCategory::Category category = (Dune::SolverCategory::Category)2; LinearOperator = Dune::OverlappingSchwarzOperator<Dune::BCRSMatrix<Dune::MatrixBlock<double, 1, 1>, std::allocator<Dune::MatrixBlock<double, 1, 1> > >, Dune::BlockVector<Dune::FieldVector<double, 1> >, Dune::BlockVector<Dune::FieldVector<double, 1> >, Dune::OwnerOverlapCopyCommunication<int> >; POrComm = Dune::OwnerOverlapCopyCommunication<int>; MatrixBlockType = Dune::MatrixBlock<double, 1, 1>; VectorBlockType = Dune::FieldVector<double, 1>; int pressureIndex = 0; Opm::ISTLSolver<MatrixBlockType, VectorBlockType, pressureIndex>::Vector = Dune::BlockVector<Dune::FieldVector<double, 1> >]’
/home/and/src/opm-simulators/build-cmake/fake-src/opm/autodiff/ISTLSolver.hpp:360:83:   required from ‘void Opm::ISTLSolver<MatrixBlockType, VectorBlockType, pressureIndex>::solve(Operator&, Opm::ISTLSolver<MatrixBlockType, VectorBlockType, pressureIndex>::Vector&, Opm::ISTLSolver<MatrixBlockType, VectorBlockType, pressureIndex>::Vector&, Comm&) const [with Operator = Dune::OverlappingSchwarzOperator<Dune::BCRSMatrix<Dune::MatrixBlock<double, 1, 1>, std::allocator<Dune::MatrixBlock<double, 1, 1> > >, Dune::BlockVector<Dune::FieldVector<double, 1> >, Dune::BlockVector<Dune::FieldVector<double, 1> >, Dune::OwnerOverlapCopyCommunication<int> >; Comm = Dune::OwnerOverlapCopyCommunication<int>; MatrixBlockType = Dune::MatrixBlock<double, 1, 1>; VectorBlockType = Dune::FieldVector<double, 1>; int pressureIndex = 0; Opm::ISTLSolver<MatrixBlockType, VectorBlockType, pressureIndex>::Vector = Dune::BlockVector<Dune::FieldVector<double, 1> >]’
/home/and/src/opm-simulators/build-cmake/fake-src/opm/autodiff/ISTLSolver.hpp:326:17:   required from ‘void Opm::ISTLSolver<MatrixBlockType, VectorBlockType, pressureIndex>::solve(Opm::ISTLSolver<MatrixBlockType, VectorBlockType, pressureIndex>::Matrix&, Opm::ISTLSolver<MatrixBlockType, VectorBlockType, pressureIndex>::Vector&, Opm::ISTLSolver<MatrixBlockType, VectorBlockType, pressureIndex>::Vector&) const [with MatrixBlockType = Dune::MatrixBlock<double, 1, 1>; VectorBlockType = Dune::FieldVector<double, 1>; int pressureIndex = 0; Opm::ISTLSolver<MatrixBlockType, VectorBlockType, pressureIndex>::Matrix = Dune::BCRSMatrix<Dune::MatrixBlock<double, 1, 1>, std::allocator<Dune::MatrixBlock<double, 1, 1> > >; Opm::ISTLSolver<MatrixBlockType, VectorBlockType, pressureIndex>::Vector = Dune::BlockVector<Dune::FieldVector<double, 1> >]’
/home/and/src/opm-simulators/build-cmake/fake-src/opm/autodiff/NewtonIterationBlackoilInterleaved.cpp:276:13:   required from ‘Opm::NewtonIterationBlackoilInterleavedImpl<np, ScalarT>::SolutionVector Opm::NewtonIterationBlackoilInterleavedImpl<np, ScalarT>::computeNewtonIncrement(const Opm::LinearisedBlackoilResidual&) const [with int np = 1; ScalarT = double; Opm::NewtonIterationBlackoilInterleavedImpl<np, ScalarT>::SolutionVector = Eigen::Array<double, -1, 1>]’
/home/and/src/opm-simulators/build-cmake/fake-src/opm/autodiff/NewtonIterationBlackoilInterleaved.cpp:202:24:   required from here
/home/and/src/opm-simulators/build-cmake/fake-src/opm/autodiff/ISTLSolver.hpp:274:68: error: no matching function for call to ‘createAMGPreconditionerPointer<Dune::Amg::CoarsenCriterion<Dune::Amg::SymmetricCriterion<Dune::BCRSMatrix<Dune::MatrixBlock<double, 1, 1>, std::allocator<Dune::MatrixBlock<double, 1, 1> > >, Dune::Amg::Diagonal<0> > > >(Dune::OverlappingSchwarzOperator<Dune::BCRSMatrix<Dune::MatrixBlock<double, 1, 1>, std::allocator<Dune::MatrixBlock<double, 1, 1> > >, Dune::BlockVector<Dune::FieldVector<double, 1> >, Dune::BlockVector<Dune::FieldVector<double, 1> >, Dune::OwnerOverlapCopyCommunication<int> >&, const double&, const Opm::MILU_VARIANT&, const Dune::OwnerOverlapCopyCommunication<int>&, std::unique_ptr<Opm::BlackoilAmg<Dune::OverlappingSchwarzOperator<Dune::BCRSMatrix<Dune::MatrixBlock<double, 1, 1>, std::allocator<Dune::MatrixBlock<double, 1, 1> > >, Dune::BlockVector<Dune::FieldVector<double, 1> >, Dune::BlockVector<Dune::FieldVector<double, 1> >, Dune::OwnerOverlapCopyCommunication<int> >, Opm::ParallelOverlappingILU0<Dune::BCRSMatrix<Dune::MatrixBlock<double, 1, 1>, std::allocator<Dune::MatrixBlock<double, 1, 1> > >, Dune::BlockVector<Dune::FieldVector<double, 1> >, Dune::BlockVector<Dune::FieldVector<double, 1> >, Dune::OwnerOverlapCopyCommunication<int> >, Dune::Amg::CoarsenCriterion<Dune::Amg::SymmetricCriterion<Dune::BCRSMatrix<Dune::MatrixBlock<double, 1, 1>, std::allocator<Dune::MatrixBlock<double, 1, 1> > >, Dune::Amg::Diagonal<0> > >, Dune::OwnerOverlapCopyCommunication<int>, 0>, std::default_delete<Opm::BlackoilAmg<Dune::OverlappingSchwarzOperator<Dune::BCRSMatrix<Dune::MatrixBlock<double, 1, 1>, std::allocator<Dune::MatrixBlock<double, 1, 1> > >, Dune::BlockVector<Dune::FieldVector<double, 1> >, Dune::BlockVector<Dune::FieldVector<double, 1> >, Dune::OwnerOverlapCopyCommunication<int> >, Opm::ParallelOverlappingILU0<Dune::BCRSMatrix<Dune::MatrixBlock<double, 1, 1>, std::allocator<Dune::MatrixBlock<double, 1, 1> > >, Dune::BlockVector<Dune::FieldVector<double, 1> >, Dune::BlockVector<Dune::FieldVector<double, 1> >, Dune::OwnerOverlapCopyCommunication<int> >, Dune::Amg::CoarsenCriterion<Dune::Amg::SymmetricCriterion<Dune::BCRSMatrix<Dune::MatrixBlock<double, 1, 1>, std::allocator<Dune::MatrixBlock<double, 1, 1> > >, Dune::Amg::Diagonal<0> > >, Dune::OwnerOverlapCopyCommunication<int>, 0> > >&, const Opm::NewtonIterationBlackoilInterleavedParameters&)’
             ISTLUtility::template createAMGPreconditionerPointer<C>( opA, relax, milu,
             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
                                                                      comm, amg, parameters_ );
                                                                      ~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/and/src/opm-simulators/build-cmake/fake-src/opm/autodiff/NewtonIterationBlackoilInterleaved.cpp:30:
/home/and/src/opm-simulators/build-cmake/fake-src/opm/autodiff/CPRPreconditioner.hpp:243:1: note: candidate: ‘template<class C, class Op, class P, class S, long unsigned int index> void Opm::ISTLUtility::createAMGPreconditionerPointer(Op&, double, const P&, std::unique_ptr<Opm::BlackoilAmg<Op, S, C, P, index> >&, const Opm::CPRParameter&)’
 createAMGPreconditionerPointer(Op& opA, const double relax, const P& comm,
 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/and/src/opm-simulators/build-cmake/fake-src/opm/autodiff/CPRPreconditioner.hpp:243:1: note:   template argument deduction/substitution failed:
In file included from /home/and/src/opm-simulators/build-cmake/fake-src/opm/autodiff/NewtonIterationBlackoilInterleaved.cpp:39:
/home/and/src/opm-simulators/build-cmake/fake-src/opm/autodiff/ISTLSolver.hpp:274:68: note:   types ‘std::unique_ptr<Opm::BlackoilAmg<Op, S, Dune::Amg::CoarsenCriterion<Dune::Amg::SymmetricCriterion<Dune::BCRSMatrix<Dune::MatrixBlock<double, 1, 1>, std::allocator<Dune::MatrixBlock<double, 1, 1> > >, Dune::Amg::Diagonal<0> > >, P, index>, std::default_delete<Opm::BlackoilAmg<Op, S, Dune::Amg::CoarsenCriterion<Dune::Amg::SymmetricCriterion<Dune::BCRSMatrix<Dune::MatrixBlock<double, 1, 1>, std::allocator<Dune::MatrixBlock<double, 1, 1> > >, Dune::Amg::Diagonal<0> > >, P, index> > >’ and ‘const Dune::OwnerOverlapCopyCommunication<int>’ have incompatible cv-qualifiers
             ISTLUtility::template createAMGPreconditionerPointer<C>( opA, relax, milu,
             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
                                                                      comm, amg, parameters_ );
                                                                      ~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/and/src/opm-simulators/build-cmake/fake-src/opm/autodiff/NewtonIterationBlackoilInterleaved.cpp:30:
/home/and/src/opm-simulators/build-cmake/fake-src/opm/autodiff/CPRPreconditioner.hpp:270:1: note: candidate: ‘template<class C, class Op, class P, class AMG> void Opm::ISTLUtility::createAMGPreconditionerPointer(Op&, double, Opm::MILU_VARIANT, const P&, std::unique_ptr<AMG>&)’
 createAMGPreconditionerPointer(Op& opA, const double relax, const MILU_VARIANT milu, const P& comm, std::unique_ptr< AMG >& amgPtr)
 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/and/src/opm-simulators/build-cmake/fake-src/opm/autodiff/CPRPreconditioner.hpp:270:1: note:   template argument deduction/substitution failed:
In file included from /home/and/src/opm-simulators/build-cmake/fake-src/opm/autodiff/NewtonIterationBlackoilInterleaved.cpp:39:
/home/and/src/opm-simulators/build-cmake/fake-src/opm/autodiff/ISTLSolver.hpp:274:68: note:   candidate expects 5 arguments, 6 provided
             ISTLUtility::template createAMGPreconditionerPointer<C>( opA, relax, milu,
             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
                                                                      comm, amg, parameters_ );
                                                                      ~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/and/src/opm-simulators/build-cmake/fake-src/opm/autodiff/NewtonIterationBlackoilInterleaved.cpp:30:
/home/and/src/opm-simulators/build-cmake/fake-src/opm/autodiff/CPRPreconditioner.hpp:299:1: note: candidate: ‘template<int pressureIndex, class Op, class P, class AMG> void Opm::ISTLUtility::createAMGPreconditionerPointer(Op&, double, Opm::MILU_VARIANT, const P&, std::unique_ptr<AMG>&)’
 createAMGPreconditionerPointer( Op& opA, const double relax, const MILU_VARIANT milu, const P& comm, std::unique_ptr< AMG >& amgPtr )
 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/and/src/opm-simulators/build-cmake/fake-src/opm/autodiff/CPRPreconditioner.hpp:299:1: note:   template argument deduction/substitution failed:
In file included from /home/and/src/opm-simulators/build-cmake/fake-src/opm/autodiff/NewtonIterationBlackoilInterleaved.cpp:39:
/home/and/src/opm-simulators/build-cmake/fake-src/opm/autodiff/ISTLSolver.hpp: In instantiation of ‘void Opm::ISTLSolver<MatrixBlockType, VectorBlockType, pressureIndex>::constructAMGPrecond(MatrixOperator&, const POrComm&, std::unique_ptr<AMG>&, std::unique_ptr<MatrixOperator>&, double, Opm::MILU_VARIANT) const [with C = Dune::Amg::CoarsenCriterion<Dune::Amg::SymmetricCriterion<Dune::BCRSMatrix<Dune::MatrixBlock<double, 1, 1>, std::allocator<Dune::MatrixBlock<double, 1, 1> > >, Dune::Amg::Diagonal<0> > >; MatrixOperator = Dune::MatrixAdapter<Dune::BCRSMatrix<Dune::MatrixBlock<double, 1, 1>, std::allocator<Dune::MatrixBlock<double, 1, 1> > >, Dune::BlockVector<Dune::FieldVector<double, 1> >, Dune::BlockVector<Dune::FieldVector<double, 1> > >; POrComm = Dune::Amg::SequentialInformation; AMG = Opm::BlackoilAmg<Dune::MatrixAdapter<Dune::BCRSMatrix<Dune::MatrixBlock<double, 1, 1>, std::allocator<Dune::MatrixBlock<double, 1, 1> > >, Dune::BlockVector<Dune::FieldVector<double, 1> >, Dune::BlockVector<Dune::FieldVector<double, 1> > >, Opm::ParallelOverlappingILU0<Dune::BCRSMatrix<Dune::MatrixBlock<double, 1, 1>, std::allocator<Dune::MatrixBlock<double, 1, 1> > >, Dune::BlockVector<Dune::FieldVector<double, 1> >, Dune::BlockVector<Dune::FieldVector<double, 1> >, Dune::Amg::SequentialInformation>, Dune::Amg::CoarsenCriterion<Dune::Amg::SymmetricCriterion<Dune::BCRSMatrix<Dune::MatrixBlock<double, 1, 1>, std::allocator<Dune::MatrixBlock<double, 1, 1> > >, Dune::Amg::Diagonal<0> > >, Dune::Amg::SequentialInformation, 0>; MatrixBlockType = Dune::MatrixBlock<double, 1, 1>; VectorBlockType = Dune::FieldVector<double, 1>; int pressureIndex = 0]’:
/home/and/src/opm-simulators/build-cmake/fake-src/opm/autodiff/ISTLSolver.hpp:168:51:   required from ‘void Opm::ISTLSolver<MatrixBlockType, VectorBlockType, pressureIndex>::constructPreconditionerAndSolve(LinearOperator&, Opm::ISTLSolver<MatrixBlockType, VectorBlockType, pressureIndex>::Vector&, Opm::ISTLSolver<MatrixBlockType, VectorBlockType, pressureIndex>::Vector&, const POrComm&, Dune::InverseOperatorResult&) const [with Dune::SolverCategory::Category category = (Dune::SolverCategory::Category)0; LinearOperator = Dune::MatrixAdapter<Dune::BCRSMatrix<Dune::MatrixBlock<double, 1, 1>, std::allocator<Dune::MatrixBlock<double, 1, 1> > >, Dune::BlockVector<Dune::FieldVector<double, 1> >, Dune::BlockVector<Dune::FieldVector<double, 1> > >; POrComm = Dune::Amg::SequentialInformation; MatrixBlockType = Dune::MatrixBlock<double, 1, 1>; VectorBlockType = Dune::FieldVector<double, 1>; int pressureIndex = 0; Opm::ISTLSolver<MatrixBlockType, VectorBlockType, pressureIndex>::Vector = Dune::BlockVector<Dune::FieldVector<double, 1> >]’
/home/and/src/opm-simulators/build-cmake/fake-src/opm/autodiff/ISTLSolver.hpp:383:13:   required from ‘void Opm::ISTLSolver<MatrixBlockType, VectorBlockType, pressureIndex>::solve(Operator&, Opm::ISTLSolver<MatrixBlockType, VectorBlockType, pressureIndex>::Vector&, Opm::ISTLSolver<MatrixBlockType, VectorBlockType, pressureIndex>::Vector&) const [with Operator = Dune::MatrixAdapter<Dune::BCRSMatrix<Dune::MatrixBlock<double, 1, 1>, std::allocator<Dune::MatrixBlock<double, 1, 1> > >, Dune::BlockVector<Dune::FieldVector<double, 1> >, Dune::BlockVector<Dune::FieldVector<double, 1> > >; MatrixBlockType = Dune::MatrixBlock<double, 1, 1>; VectorBlockType = Dune::FieldVector<double, 1>; int pressureIndex = 0; Opm::ISTLSolver<MatrixBlockType, VectorBlockType, pressureIndex>::Vector = Dune::BlockVector<Dune::FieldVector<double, 1> >]’
/home/and/src/opm-simulators/build-cmake/fake-src/opm/autodiff/ISTLSolver.hpp:333:17:   required from ‘void Opm::ISTLSolver<MatrixBlockType, VectorBlockType, pressureIndex>::solve(Opm::ISTLSolver<MatrixBlockType, VectorBlockType, pressureIndex>::Matrix&, Opm::ISTLSolver<MatrixBlockType, VectorBlockType, pressureIndex>::Vector&, Opm::ISTLSolver<MatrixBlockType, VectorBlockType, pressureIndex>::Vector&) const [with MatrixBlockType = Dune::MatrixBlock<double, 1, 1>; VectorBlockType = Dune::FieldVector<double, 1>; int pressureIndex = 0; Opm::ISTLSolver<MatrixBlockType, VectorBlockType, pressureIndex>::Matrix = Dune::BCRSMatrix<Dune::MatrixBlock<double, 1, 1>, std::allocator<Dune::MatrixBlock<double, 1, 1> > >; Opm::ISTLSolver<MatrixBlockType, VectorBlockType, pressureIndex>::Vector = Dune::BlockVector<Dune::FieldVector<double, 1> >]’
/home/and/src/opm-simulators/build-cmake/fake-src/opm/autodiff/NewtonIterationBlackoilInterleaved.cpp:276:13:   required from ‘Opm::NewtonIterationBlackoilInterleavedImpl<np, ScalarT>::SolutionVector Opm::NewtonIterationBlackoilInterleavedImpl<np, ScalarT>::computeNewtonIncrement(const Opm::LinearisedBlackoilResidual&) const [with int np = 1; ScalarT = double; Opm::NewtonIterationBlackoilInterleavedImpl<np, ScalarT>::SolutionVector = Eigen::Array<double, -1, 1>]’
/home/and/src/opm-simulators/build-cmake/fake-src/opm/autodiff/NewtonIterationBlackoilInterleaved.cpp:202:24:   required from here
```

as a quick fix I commented out the incriminating lines, but this is not a workable solution in the medium or long term. (except if it is decided that the AMG capability ought to be removed.)

**Comment by blattms** [comment]
I have problems replicating this. It works for me with g++8.2. Might depend on the DUNE version, though.

BTW: what is up with the fake-src directory underneath build-cmake? That should not be there.

**Comment by andlaus** [comment]
> I have problems replicating this. It works for me with g++8.2. Might depend on the DUNE version, though.

I'm using the Dune 2.6 branch. as said, commenting out the two offending lines makes the build succeed. if you have problem replicating this, the same offer as for OPM/opm-grid#344 stands: if you send me your SSH pubkey, I'll give you access to the machine.

> what is up with the fake-src directory underneath build-cmake? That should not be there.

that is just related to the alternative build system. it should not matter in this context since everything sans the build system infrastructure in the "real" source directory is linked into there.

**Comment by blattms** [comment]
Please double check that you did not change anything.

I will not login to your machine and am reluctant to help with customized build systems.

**Comment by blattms** [comment]
And does your ominous build system define `HAVE_UMFPACK`?

**Comment by andlaus** [comment]
```
guest@inferius:~/src> for TMP in dune-*; do (echo $TMP; cd $TMP; git pull; git diff origin/releases/2.6 | cat); done
dune-common
Already up to date.
dune-geometry
Already up to date.
dune-grid
Already up to date.
dune-istl
Already up to date.
dune-localfunctions
Already up to date.
guest@inferius:~/src> for TMP in opm-* ewoms; do (echo $TMP; cd $TMP; git pull; git diff origin/master | cat); done
opm-common
Already up to date.
opm-data
Already up to date.
opm-dune-cmake
Already up to date.
opm-grid
Already up to date.
opm-material
Already up to date.
opm-simulators
Already up to date.
ewoms
Already up to date.
```

> I will not login to your machine

in this case I recommend you to bite the bullet and set up a openSuse tumbleweed system.

> am reluctant to help with customized build systems.

this is *not* related to the build system: I just checked the same happens with the default build system. With Dune 2.5 it also does not work.

> I might have an idea. Do you have umfack and/or superlu installed and found (the build system might make a difference here!) Maybe you could upload your CMakeCache,txt as a gist?

the CMakeCache.txt for the default build system for opm-simulators with dune 2.5:

https://gist.github.com/andlaus/f4da96a5131b48280b84b14810aa13b0

**Comment by andlaus** [comment]
> And does your ominous build system define HAVE_UMFPACK?

if you mean the default build system with "ominous", then I can't say for sure. I'm pretty confident that the alternative one does.

**Comment by blattms** [comment]
Thanks for testing. You do not have UMFPack (at least according to your CMakeCache, which makes me wonder why you claim that HAVE_UMFPACK is defined). Is there any particular reason for not installing UMFPACK?

Anway I will give it a try without UMFPACK.

**Comment by blattms** [comment]
tomorrow that is...

**Comment by andlaus** [comment]
> (at least according to your CMakeCache, which makes me wonder why you claim that HAVE_UMFPACK is defined)

I did not mean say this: With the alternative build system I am confident that HAVE_UMFPACK is exactly the same as for DUNE because it uses exactly the same codepath. I'm not sure if I installed it or not, but in any case the ISTLSolver class is buggy!

**Comment by blattms** [comment]
Please try #1580

**Comment by blattms** [comment]
Just for the record: This problem should arise with all compilers if UMFPack is not found.

**Comment by blattms** [comment]
Closing in favor of #1581

**Comment by blattms** [comment]
OOps. I wanted to close the PR and not the issue.Reopening

**Comment by blattms** [comment]
Fixed with merge of #1581
